src/highlight.nit

```diff
@@ -109,3 +110,48 @@ class AbstractHighlightVisitor
 	# Low-level highlighting between 2 tokens
 	protected fun do_highlight(first_token: Token, last_token: nullable Token) is abstract
 end
+
+# Text-based highlighter that use ANSI escape sequence for colors
+class AnsiHighlightVisitor
+	super AbstractHighlightVisitor
+
+	# The produced highlighting
+	var result = new Template
+
+	redef fun do_highlight(f, l)
+	do
+		var c
+		c = f
+		while c != null do
+			if c != f then result.add(c.blank_before)
+			result.add c.ansi_colored
```

Comment:
Maybe use double dispatch so the Visitor holds the colors settings?

Reply:
in a future PR :)